Mount reload remote fs

gardouille opened the pull request:
An adaptation of similar code for local-fs.target in [debops.mount role](https://github.com/debops/debops/blob/master/ansible/roles/mount/tasks/main.yml#L62).

It also check options from `lookup/mount_options.j2` cause the user can add the systemd's automount option in several places (nfs__base_mount…, nfs__default_mount…, item.options,…).

gardouille committed:
Manage systemd's automount option for mountpoint

gardouille committed:
[doc] Add NFS systemd automount example

tasosalvas reviewed (dismissed):
An example, yay :+1:

imrejonk commented:
Thanks for the PR. Did you test your changes (with and without the --check flag)? And shouldn't 'Ensure that the NFS mount points exist' come before 'Manage NFS mount points'?

gardouille committed:
Create the path before mounting a device

gardouille commented:
@imrejonk 
Yeah, successfully test with systemd options : 
```
PLAY [Manage NFS shares] *****************************************************************************************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] *******************************************************************************************************************************************************************************************************************************************************************************************************
ok: [myhost]

TASK [nfs : Install required packages] ***************************************************************************************************************************************************************************************************************************************************************************************
ok: [myhost]

TASK [nfs : Configure NFS client] ********************************************************************************************************************************************************************************************************************************************************************************************
ok: [myhost] => (item=etc/default/nfs-common)

TASK [nfs : Ensure that the NFS mount points exist] **************************************************************************************************************************************************************************************************************************************************************************
ok: [myhost] => (item={'path': '/media/nfs/shared', 'src': 'nas.example.org:/shared', 'state': 'present', 'fstype': 'nfs4', 'default_options': False, 'options': ['defaults', 'x-systemd.automount', 'x-systemd.device-timeout=2', 'x-systemd.idle-timeout=1min', 'noatime', 'noauto']})

TASK [nfs : Manage NFS mount points] *****************************************************************************************************************************************************************************************************************************************************************************************
changed: [myhost] => (item={'path': '/media/nfs/shared', 'src': 'nas.example.org:/shared', 'state': 'present', 'fstype': 'nfs4', 'default_options': False, 'options': ['defaults', 'x-systemd.automount', 'x-systemd.device-timeout=2', 'x-systemd.idle-timeout=1min', 'noatime', 'noauto']})

TASK [nfs : Restart 'remote-fs.target' systemd unit] *************************************************************************************************************************************************************************************************************************************************************************
changed: [myhost] => (item={'src': 'nas.example.org:/shared', 'changed': True, 'name': '/media/nfs/shared', 'dump': '0', 'fstab': '/etc/fstab', 'passno': '0', 'fstype': 'nfs4', 'invocation': {'module_args': {'src': 'nas.example.org:/shared', 'name': '/media/nfs/shared', 'dump': None, 'boot': True, 'fstab': None, 'passno': None, 'fstype': 'nfs4', 'state': 'present', 'path': '/media/nfs/shared', 'backup': False, 'opts': 'defaults,x-systemd.automount,x-systemd.device-timeout=2,x-systemd.idle-timeout=1min,noatime,noauto,_netdev'}}, 'opts': 'defaults,x-systemd.automount,x-systemd.device-timeout=2,x-systemd.idle-timeout=1min,noatime,noauto,_netdev', 'failed': False, 'item': {'path': '/media/nfs/shared', 'src': 'nas.example.org:/shared', 'state': 'present', 'fstype': 'nfs4', 'default_options': False, 'options': ['defaults', 'x-systemd.automount', 'x-systemd.device-timeout=2', 'x-systemd.idle-timeout=1min', 'noatime', 'noauto']}, 'ansible_loop_var': 'item'})

PLAY RECAP *******************************************************************************************************************************************************************************************************************************************************************************************************************
myhost               : ok=6    changed=2    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0
```

and without :
```
PLAY [Manage NFS shares] *****************************************************************************************************************************************************************************************************************************************************************************************************
TASK [Gathering Facts] *******************************************************************************************************************************************************************************************************************************************************************************************************
ok: [myhost]

TASK [nfs : Install required packages] ***************************************************************************************************************************************************************************************************************************************************************************************
ok: [myhost]

TASK [nfs : Configure NFS client] ********************************************************************************************************************************************************************************************************************************************************************************************
ok: [myhost] => (item=etc/default/nfs-common)

TASK [nfs : Ensure that the NFS mount points exist] **************************************************************************************************************************************************************************************************************************************************************************ok: [myhost] => (item={'path': '/media/nfs/shared', 'src': 'nas.example.org:/shared', 'state': 'present', 'fstype': 'nfs4', 'default_options': False})

TASK [nfs : Manage NFS mount points] *****************************************************************************************************************************************************************************************************************************************************************************************changed: [myhost] => (item={'path': '/media/nfs/shared', 'src': 'nas.example.org:/shared', 'state': 'present', 'fstype': 'nfs4', 'default_options': False})

PLAY RECAP *******************************************************************************************************************************************************************************************************************************************************************************************************************
myhost               : ok=5    changed=1    unreachable=0    failed=0    skipped=2    rescued=0    ignored=0
```

Also create mountpoint before mounting the device as pointed :)

drybjed reviewed (approved):
Nicely done, thanks. I'll backport this as well.
